Refactor menu endpoint JSON response

diraulo opened the pull request:
**PT: https://www.pivotaltracker.com/story/show/95106938**

responses from `/v1/menus` will now have the following structure

``` javascript
{
  "date_range": "2015-05-04..2015-05-10",
  "menus": [
    {
      "id": 2,
      "title": "Monday",
      "start_date": "2015-05-04",
      "end_date": "2015-05-04"
      "uri": "http://localhost:3000/v1/menus/2",
      "item_count": 2,
      "items": [
        {
          "id": 1,
          "item": "Kyckling",
          "price": "20.0",
          "description": "Lorem ipsum"
          "ingredients": "something tasty, ...",
        },
        {
          "id": 2,
          "item": "Biff",
          "price": "15.0",
          "description": "Lorem ipsum..."
          "ingredients": "beef, ...",
        }
      ]
    }
  ]
}
```

Disable caching => **PT: https://www.pivotaltracker.com/story/show/94062700**

diraulo committed:
Refactor menu endpoint JSON response

diraulo committed:
disable caching on menu jbuilder view

razorcd commented on spec/requests/api/v1/menu_spec.rb:
```diff
@@ -19,134 +18,132 @@
+                                  'items' => [
+                                    {
+                                      'id' => @menu_items[0].id,
+                                      'name' => @menu_items[2].name,
```
`@menu_items[2].name` instead of `@menu_items[0].name` and the tests are passing? Something is wrong.

diraulo replied:
I know why the tests are passing. It has to do with the factories. The way they're defined right now will not work properly with Faker. It generates the same name for every factories. I'll fix this now. Thanks @razorcd

diraulo committed:
Fix menu & menu_items factories

- When using `Faker` to generate random values, it is essential
to wrap them in curly braces, otherwise it generates the same
values for every object.
- Fix typo in `menu_spec`
